### ISSUE TITLE
clang++.exe in MSVC prompt uses MSVC-specific arguments

### ISSUE THREAD

**Issue by oneseedfruit** [godotengine/godot#8412]
**Operating system or device - Godot version:**
Windows 10. Godot 2.1.3-stable

**Issue description:**
I am trying to compile for Android at commit d8ae244d51ed157d96346c510baeb637239e3200 and I'm getting:

```
clang++.exe: error: no such file or directory: '/W4'

scons: *** [platform\android\os_android.os] Error 1

clang++.exe: error: no such file or directory: '/W4'

scons: *** [platform\android\godot_android.os] Error 1
scons: building terminated because of errors.
```

full log: https://gist.github.com/ranmaru90/f0a7e94648a20d4013cde7c0216aaa3d

Compiled with Microsoft Visual C++ Compiler 14.0 (amd64). Python 2.7.13, and I have the latest Android SDK and NDK. 

However, it compiled successfully when I switched to TDM-GCC MinGW 64. 

A separate not-sure-if-it's-an-issue: Compiling for Windows using TDM-GCC MinGW 64 works too. However if I run the compiled Godot and switch to the "Script" tab, if I open many scripts at a time, then try to close them all, Godot will crash. This doesn't happen if I compiled with MSVC 14.0.

**Steps to reproduce:**
1. Be at commit d8ae244d51ed157d96346c510baeb637239e3200
2. Be on Windows 10, use MSVC 14.0 (amd64).
3. scons platform=android target=release_debug tools=no -j 2 && scons platform=android target=release tools=no -j 2

**Comment by vnen** [comment]
You shouldn't build with a MSVC prompt. MSVC is not used to build for Android.

I'm able build master (b7ce160) for Android on Windows 10 just fine with the latest NDK (r14b).

**Comment by ranmaru90** [comment]
I see, but why was I able to build for Android using MSVC before this?

**Comment by akien-mga** [comment]
That's because of the way we detect if we need to use MSVC command line arguments which are different from those shared by Clang and GCC: https://github.com/godotengine/godot/blob/2.1/SConstruct#L281

Here you use the MSVC prompt but it actually runs Clang, so that logic doesn't work. If anyone knows a better way to detect the actual compiler used, would be nice :)

**Comment by vnen** [comment]
This should only be applied when building for Windows and UWP.

**Comment by akien-mga** [comment]
> This should only be applied when building for Windows and UWP.

Yes, but also only when building with MSVC, not MinGW (or Clang++.exe apparently). Looking at the above, I guess it might even be possible to build Windows binaries with Clang in the MSVC prompt... So we need some bullet proof logic and maybe put it in methods.py so that it can be reused in other places.

**Comment by kubecz3k** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about its current status here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
This should be fixed by now, we're using SCons' built-in detection of MSVC to set the proper `env.msvc` bool.
